### PR TITLE
REGRESSION(264584@main): Web Extension debug logging is not being logged.

### PR DESCRIPTION
**Pull request by xeenon**
#### 8e01542d58ad3d5c7370245d7cd460066215c3d1
<pre>
REGRESSION(264584@main): Web Extension debug logging is not being logged.
<a href="https://webkit.org/b/272918">https://webkit.org/b/272918</a>
<a href="https://rdar.apple.com/problem/126708221">rdar://problem/126708221</a>

Reviewed by Per Arne Vollan.

os_log_set_hook takes a minimum log type, so set this to OS_LOG_TYPE_DEBUG for
debug builds. Also only return early for debug and info types in release builds.

This makes sure debug builds get all log message types. While release builds stay
mostly the same as before. However, info logs are now also filtered out, which was
the expected behavior before, but info logs were still getting sent.

* Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:
(WebKit::registerLogHook):

Canonical link: <a href="https://commits.webkit.org/277700@main">https://commits.webkit.org/277700@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/8eb85e960e308d9b4d36b3b47424c9ea49330fec

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/48331 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/27543 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/51285 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/51019 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/44396 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/33479 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/25064 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/39486 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/48913 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/25230 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/41739 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/20630 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/22710 "Passed tests") | | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/6387 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/44666 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/36/builds/43377 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/52923 "Built successfully") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/23378 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/50/builds/19719 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/46818 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/24643 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/41928 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/45732 "Passed tests") | 
| | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/25448 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/6873 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/24366 "Built successfully") | | |

### CODE REVIEW

**Review by pvollan — APPROVED** [review]
R=me.

File: Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm [hunk 1]
```diff
@@ -821,16 +821,27 @@ static void registerLogHook()
...
 
         if (msg->buffer_sz > 1024)
             return;
 
-        // Skip debug logs in non-internal builds.
-        if (type == OS_LOG_TYPE_DEBUG)
+#ifdef NDEBUG
+        // Don't send messages with types we don't want to log in release. Even though OS_LOG_TYPE_DEFAULT is the minimum,
+        // the hook will be called for other subsystems with debug and info types.
+        if (type & (OS_LOG_TYPE_DEBUG | OS_LOG_TYPE_INFO))
```
Comment (pvollan) [inline Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:842]: I think we want to send OS_LOG_TYPE_INFO?

Comment (pvollan) [reply Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:842]: Yes, I think we only should return if type is OS_LOG_TYPE_DEBUG here.

Comment (xeenon) [reply Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm:842]: This only rejects `OS_LOG_TYPE_INFO` logs from other subsystems, with `minimumType = OS_LOG_TYPE_DEFAULT` the hook is not called for WebKit's subsystem for `OS_LOG_TYPE_INFO`.

**Pull request merged**